verifiable-db/src/query/batching/circuits/api.rs

```diff
@@ -0,0 +1,1042 @@
+    /// - `children`: data about the children of the given node
+    pub fn new(
+        node_info: NodeInfo,
+        path: Vec<(NodeInfo, ChildPosition)>,
+        siblings: Vec<Option<HashOutput>>,
+        children: [Option<NodeInfo>; 2],
+    ) -> Self {
+        Self {
+            node_info,
+            path,
+            siblings,
+            children,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Default, PartialEq, Eq)]
+/// Data structure containing the information about the paths in both the rows tree
+/// and the index tree for a generic row of our tables
+pub struct RowPath {
+    pub(crate) row_tree_path: TreePathInputs,
+    /// Info about the node of the index tree storing the rows tree containing the row
+    pub(crate) index_tree_path: TreePathInputs,
+}
+
+impl RowPath {
+    /// Instantiate a new instance of `RowPath` for a given proven row from the following input data:
+    /// - `row_node_info`: data about the node of the row tree storing the row
+    /// - `row_tree_path`: data about the nodes in the path of the rows tree for the node storing the row;
+    ///     The `ChildPosition` refers to the position of the previous node in the path as a child of the current node
+    /// - `row_path_siblings`: hash of the siblings of the node in the rows tree path (except for the root)
+    /// - `row_node_children`: data about the children of the node of the row tree storing the row
+    /// - `index_node_info`: data about the node of the index tree storing the rows tree containing the row
+    /// - `index_tree_path`: data about the nodes in the path of the index tree for the index_node;
+    ///     The `ChildPosition` refers to the position of the previous node in the path as a child of the current node
+    /// - `index_path_siblings`: hash of the siblings of the nodes in the index tree path (except for the root)
+    /// - `index_node_children`: data about the children of the index_node
+    pub fn new(
+        row_node_info: NodeInfo,
+        row_tree_path: Vec<(NodeInfo, ChildPosition)>,
+        row_path_siblings: Vec<Option<HashOutput>>,
+        row_node_children: [Option<NodeInfo>; 2],
+        index_node_info: NodeInfo,
+        index_tree_path: Vec<(NodeInfo, ChildPosition)>,
+        index_path_siblings: Vec<Option<HashOutput>>,
+        index_node_children: [Option<NodeInfo>; 2],
+    ) -> Self {
+        let row_path = TreePathInputs::new(
+            row_node_info,
+            row_tree_path,
+            row_path_siblings,
+            row_node_children,
+        );
+        let index_path = TreePathInputs::new(
+            index_node_info,
+            index_tree_path,
+            index_path_siblings,
+            index_node_children,
+        );
+        Self {
+            row_tree_path: row_path,
+            index_tree_path: index_path,
+        }
+    }
+}
+
+pub struct RowWithPath {
+    pub(crate) cells: RowCells,
+    pub(crate) path: RowPath,
+}
```

Comment:
Done in commit https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/408/commits/b2571453d04e51ebe711251f0a54e2ce7c93ef16. I have also renamed the struct to `RowInput`, because it's actually the only row-specific input needed to generate the proof